app/Http/Controllers/Admin/LanguageSettingController.php

```diff
@@ -43,7 +43,39 @@ public function update(Request $request): RedirectResponse
             return History::redirectBack();
         } catch (NotValidSettingRecordException $e) {
             SystemMessageService::addErrorMessage('system_messages.language.invalid_record');
-            return redirect()->back()->withInput();
+            return redirect('language.edit');
+        }
+    }
+
+    /**
+     * @role(super_user, cms_manager, acc_manager)
+     */
+    public function add(): View
+    {
+        $available_languages = LanguageSettingService::getAvailableChoices();
+        return view('admin.pages.language.create',
+            [
+                'available_languages' => $available_languages
+            ]
+        );
+    }
+
+    /**
+     * @role(super_user, cms_manager, acc_manager)
+     */
+    public function store(Request $request): RedirectResponse
+    {
+        try {
+            LanguageSettingService::setOne(
+                $request->input('language_id'),
+                $request->has('is_enabled'),
+                $request->has('is_default'),
+            );
+            SystemMessageService::addSuccessMessage('زبان با موفقیت اضافه شد');
```

Comment:
For this section, don't use the Persian message hard coded in the controller.
you can pass 'messages.language.language_addes_successfully' and then add the language_added_successfully to the message.php file in the resources/lang directory.